src/neo4j_graphrag/experimental/pipeline/pipeline.py

```diff
@@ -182,40 +189,100 @@ def show_as_dict(self) -> dict[str, Any]:
+        self, path: str, layout: str = "force", hide_unused_outputs: bool = True
     ) -> Any:
-        G = self.get_pygraphviz_graph(hide_unused_outputs)
-        G.layout(layout)
-        G.draw(path)
+        """Draw the pipeline graph using neo4j-viz.
 
-    def get_pygraphviz_graph(self, hide_unused_outputs: bool = True) -> pgv.AGraph:
-        if pgv is None:
+        Args:
+            path (str): Path to save the visualization. If the path ends with .html, it will save an HTML file.
+                Otherwise, it will save a PNG image.
+            layout (str): Layout algorithm to use. Default is "force".
+            hide_unused_outputs (bool): Whether to hide unused outputs. Default is True.
+
+        Returns:
+            Any: The visualization object.
+        """
+        G = self.get_neo4j_viz_graph(hide_unused_outputs)
+        if path.endswith(".html"):
+            # Save as HTML file
+            with open(path, "w") as f:
+                f.write(G.render()._repr_html_())
+        else:
+            # For other formats, we'll use the render method and save the image
+            G.render()
+            # Note: neo4j-viz doesn't support direct saving to image formats
+            # If image format is needed, consider using a screenshot or other methods
+            with open(path, "w") as f:
+                f.write(G.render()._repr_html_())
+
+    def get_neo4j_viz_graph(self, hide_unused_outputs: bool = True) -> NeoVizGraph:
+        """Create a neo4j-viz visualization graph from the pipeline.
+
+        Args:
+            hide_unused_outputs (bool): Whether to hide unused outputs. Default is True.
+
+        Returns:
+            NeoVizGraph: The neo4j-viz visualization graph.
+        """
+        if not HAS_NEO4J_VIZ:
             raise ImportError(
-                "Could not import pygraphviz. "
-                "Follow installation instruction in pygraphviz documentation "
-                "to get it up and running on your system."
+                "Could not import neo4j-viz. "
+                "Install it with 'pip install neo4j-viz'."
             )
         self.validate_parameter_mapping()
-        G = pgv.AGraph(strict=False, directed=True)
-        # create a node for each component
+
+        nodes = []
+        relationships = []
+        node_ids = {}
+        node_counter = 0
+
+        # Create nodes for each component
         for n, node in self._nodes.items():
             comp_inputs = ",".join(
                 f"{i}: {d['annotation']}"
                 for i, d in node.component.component_inputs.items()
             )
-            G.add_node(
-                n,
-                node_type="component",
-                shape="rectangle",
-                label=f"{node.component.__class__.__name__}: {n}({comp_inputs})",
+            node_ids[n] = node_counter
+            nodes.append(
+                Node(
+                    id=node_counter,
```

Comment:
According to [the doc](https://neo4j.com/docs/nvl-python/preview/api-reference/node.html), it looks like `id` can be a string? That could simplify the id and node matching part.